blake2b/blake2b.go

```diff
@@ -0,0 +1,61 @@
+package blake2b
+
+import (
+	"hash"
+
+	"golang.org/x/crypto/blake2b"
+)
+
+type state struct {
+	// the underlying blake2b_128 hasher
+	hasher hash.Hash
+
+	// the key we need to append to the finalized hash digest (see Sum())
+	key []byte
+}
+
+func (s *state) Write(p []byte) (n int, err error) {
+	// save the key for later use in Sum()
+	s.key = append(s.key, p...)
+
+	return s.hasher.Write(p)
+}
+
+func (s *state) Sum(b []byte) []byte {
+	// append key to final hash digest
+	return append(b, append(s.hasher.Sum(nil), s.key...)...)
+}
+
+func (s *state) Reset() {
+	s.hasher.Reset()
+}
+
+func (s *state) Size() int {
+	return s.hasher.Size()
```

Comment:
This is not accurate, since `s.key` is appended to the `sum` of `hasher`.

```suggestion
	return s.hasher.Size() + len(key)
```

Reply:
Good catch, will push a fix.

Reply:
fixed in latest upload